cmd/ocitools/generate.go

```diff
@@ -305,61 +311,73 @@ func setupSpec(g *generate.Generator, context *cli.Context) error {
 		}
 	}
 
-	var sd string
-	var sa, ss []string
-
-	if context.IsSet("seccomp-default") {
-		sd = context.String("seccomp-default")
+	err := addSeccomp(g, context)
+	if err != nil {
+		return err
 	}
 
-	if context.IsSet("seccomp-arch") {
-		sa = context.StringSlice("seccomp-arch")
-	}
+	return nil
+}
 
-	if context.IsSet("seccomp-syscalls") {
-		ss = context.StringSlice("seccomp-syscalls")
+func addSeccomp(g *generate.Generator, context *cli.Context) error {
+
+	// Set the DefaultAction of seccomp
+	seccompDefault := context.String("seccomp-default")
+	err := g.SetDefaultSeccompAction(seccompDefault)
+	if err != nil {
+		return err
 	}
 
-	if sd == "" && len(sa) == 0 && len(ss) == 0 {
-		return nil
+	// Add the additional architectures permitted to be used for system calls
+	seccompArch := context.String("seccomp-arch")
+	architectureArgs := strings.Split(seccompArch, ",")
+	err = g.SetSeccompArchitectures(architectureArgs)
```

Comment:
Rather than passing arguments for parsing in the library, we should pass an option struct here.